src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4759,10 +4764,10 @@ final boolean complete(LogonCommand logonCommand, TDSReader tdsReader) throws SQ
         int dataLen = 0;
 
         // Denali --> TDS 7.4, Katmai (10.0) & later 7.3B, Prelogin disconnects anything older
-        tdsVersion = serverMajorVersion >= 11 ? TDS.VER_DENALI
-                                              : serverMajorVersion >= 10 ? TDS.VER_KATMAI
-                                                                         : serverMajorVersion >= 9 ? TDS.VER_YUKON
-                                                                                                   : TDS.VER_UNKNOWN;
+        tdsVersion = (serverMajorVersion >= 11) ? TDS.VER_DENALI
+                                                : (serverMajorVersion >= 10) ? TDS.VER_KATMAI
```

Comment:
Needs precise bracketing for nested ternary operations.

Reply:
revert cos it looks ugly now